lib/datasource/sbt-package/index.ts

```diff
@@ -64,6 +75,66 @@ export async function resolvePackageReleases(
   return null;
 }
 
+export function getLatestVersion(versions: string[]): string | null {
+  if (versions && versions.length) {
+    return versions.reduce((latestVersion, version) =>
+      compare(version, latestVersion) === 1 ? version : latestVersion
+    );
+  }
+  return null;
+}
+
+export async function getUrls(
+  searchRoot: string,
+  artifactDirs: string[],
+  version: string
+): Promise<Partial<ReleaseResult>> {
+  const result: Partial<ReleaseResult> = {};
+
+  if (!artifactDirs || !artifactDirs.length) {
+    return result;
+  }
+
+  if (!version) {
+    return result;
+  }
+
+  for (const artifactDir of artifactDirs) {
+    const [artifact] = artifactDir.split('_');
+    const pomFileNames = [
+      `${artifactDir}-${version}.pom`,
+      `${artifact}-${version}.pom`,
+    ];
+
+    for (const pomFileName of pomFileNames) {
+      const pomUrl = `${searchRoot}/${artifactDir}/${version}/${pomFileName}`;
+      const content = await downloadHttpProtocol(pomUrl, 'sbt');
+
+      if (content) {
+        const pomXml = new XmlDocument(content);
+
+        const homepage = pomXml.valueWithPath('url');
+        if (homepage) {
+          result.homepage = homepage;
+        }
+
+        const sourceUrl = pomXml.valueWithPath('scm.url');
+        if (sourceUrl) {
+          result.sourceUrl = sourceUrl
+            .replace(/^scm:/, '')
+            .replace(/^git:/, '')
+            .replace(/^git@github.com:/, 'https://github.com/')
+            .replace(/[.]git$/, '');
+        }
```

Comment:
I think we should break the loop here if `homepage` and `sourceUrl` are found
```suggestion
          if (result.homepage && result.sourceUrl) {
            return result;
          }
        }
```

Reply:
For now, the loop breaks anyway if we find POM file even if `homepage` or `sourceUrl` is missing.  I guess your suggestion is to continue the loop while one of them is missing.  So, I think [this `return` statement](https://github.com/renovatebot/renovate/pull/6756/files#diff-70567177db680f9b3330add88ac8ec6eR130) is the one that should be replaced by your suggested code.  Am I right?

Reply:
Ups, oversight. we don't want to change that, so ignore this.